### ISSUE TITLE
controllers hardly use realtime_tools, they should

### ISSUE THREAD

**Issue by bmagyar** [ros-controls/ros2_controllers#302]
realtime_tools was not ported really well which led to little adoption of it in the new controllers. 
 
Once we have https://github.com/ros-controls/realtime_tools/issues/81 completed, the controllers should be reviewed & refactored to make use of realtime_tools.

**Comment by christophfroehlich** [comment]
https://github.com/ros-controls/realtime_tools/issues/279

**Comment by SouriRishik** [comment]
Hi @christophfroehlich , Should I work on this issue. Or should I take up any other current issues which need to fixed?

**Comment by christophfroehlich** [comment]
afaik the only controller not using the latest box/queue is the JTC. I started working on it but never had the time to fix the failing tests https://github.com/ros-controls/ros2_controllers/pull/1480
You want to give it a try?

The other thing is the realtime publisher, there exists a stale PR https://github.com/ros-controls/realtime_tools/pull/469 you could start from.

**Comment by SouriRishik** [comment]
yes I could definately try that. Let me know how you want me to proceed

**Comment by christophfroehlich** [comment]
Choose one, and go for it ;)

**Comment by SouriRishik** [comment]
Hi @christophfroehlich, meanwhile until the PR is merged should I work on anything else?

**Comment by christophfroehlich** [comment]
How about the linked PR for the RealtimePublisher?

**Comment by SouriRishik** [comment]
Yes, I can work on that. I'll take a look at the PR https://github.com/ros-controls/realtime_tools/pull/469

**Comment by SouriRishik** [comment]
Done @christophfroehlich, opened a PR: ros-controls/realtime_tools#573
